### PR TITLE
Add OCCA to Travis CI

### PR DESCRIPTION
**Pull request by jeremylt**
This PR adds OCCA, specifically `/cpu/occa` and `/omp/occa` to Travis CI. This PR addresses Issue #94.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -39,3 +42,5 @@ cache:
   ccache: true
   directories:
   - $HOME/install
+  - $HOME/mfem
+  - $HOME/occa
```
Comment (jedbrown) [inline .travis.yml]: Do we want these cached instead of relying on ccache?

Comment (jedbrown) [reply .travis.yml]: With PETSc, we are using a release and have logic to skip the install entirely (no configure, no make, no install; just use `$HOME/install/petsc-3.9.3`). We do not have that for MFEM or OCCA, both of which are built in-source using just `make`. We should delete these two lines. The builds will still be fast due to ccache.

Comment (jeremylt) [reply .travis.yml]: Updated. The run time is faster that I expected this time.

File: .travis.yml [hunk 2]
```diff
@@ -17,6 +17,9 @@ env:
 install:
   - git clone --depth 1 https://github.com/mfem/mfem.git
   - make -C mfem -j2 serial MFEM_CXXFLAGS=-O
+  - git clone --depth 1 https://github.com/libocca/occa.git
+  - make -C occa -j2
+  - export OCCA_DIR=$PWD/occa LD_LIBRARY_DIR+=":${PWD}/lib"
```
Comment (jedbrown) [inline .travis.yml]: What is `LD_LIBRARY_DIR`?

Comment (jedbrown) [reply .travis.yml]: Also, that path doesn't look like it is anything. I expect you can delete that part.

Comment (jeremylt) [reply .travis.yml]: Its in the OCCA build instructions. I sometimes have trouble locally when I skip it.

Comment (jeremylt) [reply .travis.yml]: It's where `libocca.so` lives

Comment (jedbrown) [reply .travis.yml]: `LD_LIBRARY_PATH` is in the build instructions. (`LD_LIBRARY_DIR` isn't a thing as far as I know.) LibCEED uses RPATH to link so the environment variable isn't needed. The library is in `$OCCA_DIR/lib`, not `$PWD/lib`.

Comment (jeremylt) [reply .travis.yml]: Should be all good now. Build times are much more reasonable today

**Pull request merged**